src/unpack.rs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2021 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use libc::timeval;
+use std::collections::HashMap;
+use std::ffi::CString;
+use std::fs;
+use std::io;
+use std::path::Path;
+use tar::Archive;
+
+/// Unpack the contents of tarball to the destination path
+pub fn unpack(input: Vec<u8>, destination: &Path) -> Result<()> {
+    let mut archive = Archive::new(input.as_slice());
+
+    if destination.exists() {
+        return Err(anyhow!(
+            "unpack destination {:?} already exists",
+            destination
+        ));
+    }
+
+    fs::create_dir_all(destination)?;
+
+    let mut dirs: HashMap<CString, [timeval; 2]> = HashMap::default();
+    for file in archive.entries()? {
+        let mut file = file?;
+        file.unpack_in(destination)?;
+
+        // tar-rs crate only preserve timestamps of files,
+        // symlink file and directory are not covered.
```

Comment:
Is this https://github.com/alexcrichton/tar-rs/pull/217? If so, I'd consider adding a URL in the comments.